src/ol_infrastructure/applications/learn_ai/__main__.py

```diff
@@ -0,0 +1,1236 @@
+
+from bridge.lib.constants import FASTLY_A_TLS_1_3
+from bridge.lib.magic_numbers import (
+    AWS_RDS_DEFAULT_DATABASE_CAPACITY,
+    DEFAULT_HTTPS_PORT,
+    DEFAULT_NGINX_PORT,
+    DEFAULT_POSTGRES_PORT,
+    DEFAULT_REDIS_PORT,
+    DEFAULT_UWSGI_PORT,
+    ONE_MEGABYTE_BYTE,
+)
+from bridge.secrets.sops import read_yaml_secrets
+from bridge.settings.github.team_members import DEVOPS_MIT
+from ol_infrastructure.components.aws.cache import OLAmazonCache, OLAmazonRedisConfig
+from ol_infrastructure.components.aws.database import OLAmazonDB, OLPostgresDBConfig
+from ol_infrastructure.components.services.vault import (
+    OLVaultDatabaseBackend,
+    OLVaultK8SDynamicSecretConfig,
+    OLVaultK8SResources,
+    OLVaultK8SResourcesConfig,
+    OLVaultK8SSecret,
+    OLVaultK8SStaticSecretConfig,
+    OLVaultPostgresDatabaseConfig,
+)
+from ol_infrastructure.lib.aws.cache_helper import CacheInstanceTypes
+from ol_infrastructure.lib.aws.eks_helper import (
+    check_cluster_namespace,
+    default_psg_egress_args,
+    get_default_psg_ingress_args,
+    setup_k8s_provider,
+)
+from ol_infrastructure.lib.aws.iam_helper import IAM_POLICY_VERSION, lint_iam_policy
+from ol_infrastructure.lib.aws.rds_helper import DBInstanceTypes
+from ol_infrastructure.lib.fastly import (
+    build_fastly_log_format_string,
+    get_fastly_provider,
+)
+from ol_infrastructure.lib.ol_types import AWSBase
+from ol_infrastructure.lib.pulumi_helper import parse_stack
+from ol_infrastructure.lib.stack_defaults import defaults
+from ol_infrastructure.lib.vault import setup_vault_provider
+
+aws_account = get_caller_identity
+
+stack_info = parse_stack()
+
+cluster_stack = StackReference(f"infrastructure.aws.eks.applications.{stack_info.name}")
+dns_stack = StackReference("infrastructure.aws.dns")
+monitoring_stack = StackReference("infrastructure.monitoring")
+network_stack = StackReference(f"infrastructure.aws.network.{stack_info.name}")
+policy_stack = StackReference("infrastructure.aws.policies")
+vault_stack = StackReference(f"infrastructure.vault.operations.{stack_info.name}")
+vector_log_proxy_stack = StackReference(
+    f"infrastructure.vector_log_proxy.operations.{stack_info.name}"
+)
+
+apps_vpc = network_stack.require_output("applications_vpc")
+k8s_pod_subnet_cidrs = apps_vpc["k8s_pod_subnet_cidrs"]
+learn_ai_environment = f"applications-{stack_info.env_suffix}"
+
+aws_config = AWSBase(
+    tags={"OU": "operations", "Environment": learn_ai_environment},
+)
+learn_ai_config = Config("learn_ai")
+vault_config = Config("vault")
+
+setup_vault_provider(stack_info)
+fastly_provider = get_fastly_provider()
+github_provider = github.Provider(
+    "github_provider",
+    owner=read_yaml_secrets(Path("pulumi/github_provider.yaml"))["owner"],
+    token=read_yaml_secrets(Path("pulumi/github_provider.yaml"))["token"],
+)
+
+k8s_global_labels = {
+    "ol.mit.edu/stack": stack_info.full_name,
+    "ol.mit.edu/service": "learn-ai",
+}
```

Comment:
This is something we should probably turn into a Pydantic object similar to `AWSBase` so that we can have consistent labels/tags especially for cost allocation via Vantage. (in a follow-up PR, not a blocker for this work)

Reply:
https://github.com/mitodl/ol-infrastructure/issues/2946